starlite/asgi.py

```diff
@@ -20,6 +29,18 @@
     from starlite.types import LifeCycleHandler
 
 
+# Sentinel object to represent a path param in the route map
+class _PathParam:
+    def __repr__(self) -> str:  # pragma: no cover
+        return "*"
+
+
+_path_param = _PathParam()
+
+RouteMapDict = MutableMapping[Union[str, _PathParam], Any]
```

Comment:
A better name IMO would be `RouteMapNode`. Also, any reason not to use `Dict[]` as a type?